pkg/kubernetes/watcher.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"fmt"
+	"regexp"
+	"sync"
+	"time"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+
+	collector_metric "github.com/sustainable-computing-io/kepler/pkg/collector/metric"
+	"github.com/sustainable-computing-io/kepler/pkg/config"
+)
+
+const (
+	informerTimeout = time.Minute
+	podResourceType = "pods"
+)
+
+var (
+	regexReplaceContainerIDPrefix = regexp.MustCompile(`.*//`)
+	IsWatcherEnabled              = false
+	managedPods                   = make(map[string]bool)
+)
+
+type ObjListWatcher struct {
+	// Lock to syncronize the collector update with the watcher
+	Mx *sync.Mutex
+
+	k8sCli       *kubernetes.Clientset
```

Comment:
you only need the clientset to intiliaze the kubernetes list watcher so there shouldn't be any needs to keep it in the struct

Reply:
Not now, but we might use it later to run benchmarks for model training.